### PR TITLE
Fixes for zimwiki writer

### PR DESCRIPTION
**Pull request by alexivkin**
Improvements for ZimWiki's source view and table markup.

### CODE REVIEW

File: src/Text/Pandoc/Writers/ZimWiki.hs [hunk 1]
```diff
@@ -44,20 +44,21 @@ import Data.Default (Default(..))
...
   }
 
 instance Default WriterState where
-  def = WriterState { stItemNum = 1, stIndent = "" }
+  def = WriterState { stItemNum = 1, stIndent = "", stInTable = False }
 
 -- | Convert Pandoc to ZimWiki.
 writeZimWiki :: PandocMonad m => WriterOptions -> Pandoc -> m String
-writeZimWiki opts document = return $ evalState (pandocToZimWiki opts document) (WriterState 1 "")
+writeZimWiki opts document = return $ evalState (pandocToZimWiki opts document) (WriterState 1 "" False)
```
Comment (tarleb) [inline src/Text/Pandoc/Writers/ZimWiki.hs]: The WriterState seems to be def, maybe it would make sense to use that instead?

Comment (alexivkin) [inline src/Text/Pandoc/Writers/ZimWiki.hs]: Def?

Comment (tarleb) [inline src/Text/Pandoc/Writers/ZimWiki.hs]: The `def` value from the `Default` type class.

Comment (alexivkin) [inline src/Text/Pandoc/Writers/ZimWiki.hs]: I have no idea how to do that. Are there any existing examples in the code?

Comment (mb21) [inline src/Text/Pandoc/Writers/ZimWiki.hs]: I think what @tarleb is proposing is simply: `evalState (pandocToZimWiki opts document) def`

**Comment by alexivkin** [comment]
Fixed to def and added 2 more fixes. Not shure why travis checks never complete.

**Comment by mb21** [comment]
@alexivkin e.g. [this travis job](https://travis-ci.org/jgm/pandoc/jobs/201327537) is failing because of the zimwiki test. Try `stack test` to run the tests on your local system... and update the `/test/writer.zimwiki` file if necessary...

**Comment by alexivkin** [comment]
updated the tests

**Comment by alexivkin** [comment]
The tests work for zim, but fail on Latex - completely unrelated to my code. It's something in the previous main branch commits that triggered it.

**Comment by jgm** [comment]
The rebase has gotten pretty hairy.  You added this at a bad time, with lots of churn in the code base.  Sorry this has been difficult.

Suggestion:

1.  remove this PR
2.  copy your src/Text/Pandoc/Writers/ZimWiki.hs, test/writer.zimwiki, and test/table.zimwiki to a safe place.
3.  check out current pandoc master
4.  copy the three files backed up above to their appropriate places in master
5.  git add and commit them.  **Be sure to include a full commit message, describing the changes you've made..."fixed the writer" is not enough.**  I'll need this for the changelog.
6.  stack test
7.  assuming all goes well, create a new PR.  This will be a single clean commit on top of master, with a good description of the changes, and should be mergeable.

**Comment by alexivkin** [comment]
Done. https://github.com/jgm/pandoc/pull/3446
Should have I committed to a branch, instead of master, to avoid trying to catch up with the head?

**Comment by mb21** [comment]
great!

> Should have I committed to a branch, instead of master, to avoid trying to catch up with the head?

commiting to a branch frees the master branch of your own repo to work on other features while you wait for jgm to merge your pull request into the official master. you always have to catch up with master (best to rebase, not merge), but this time you were just unlucky that there were a lot of changes in the meantime.

**Comment by jgm** [comment]
I don't know, perhaps that would have made things easier,
but I think the main problem is the large number of changes
to HEAD in this period.

+++ Alex Ivkin [Feb 14 17 17:01 ]:

**Pull request closed without merging**